Automattic/wp-calypso
# Jetpack Licensing Portal: Add valid payment method validation for license assignment route

#### Changes proposed in this Pull Request

* Add payment method validation to protect license assignment route from direct access.

#### Testing instructions

* Make sure to have a Partner Account 👉🏻 PCYsg-zPi-p2
* Issue some license
* If you don't have any valid payment methods you will be redirected to the payment methods addition page
* If you try to access the license assignment page directly, you will be redirected to the payment methods addition page

> The `has_valid_payment_method` key is `false` for the current partner without a valid payment method on file. If you want to to test that it does not redirect if the user does have a valid payment method, please contact @avalon-team to help to toggle that flag.
